### PR TITLE
[GTests Refactoring] Removing MIOPEN_TEST_ALL and MIOPEN_TEST_FLOAT_ARG env variables. Part 1.

### PR DESCRIPTION
**Pull request by Vsevolod1983**
Part 1.

Base branch for this PR is GTests_Refactoring_Integration_Branch

### CODE REVIEW

**Comment by junliume** [comment]
Is the target branch `[GTests_Refactoring_Integration_Branch](https://github.com/ROCm/MIOpen/tree/GTests_Refactoring_Integration_Branch)` ?

**Comment by Vsevolod1983** [comment]
> Is the target branch `[GTests_Refactoring_Integration_Branch](https://github.com/ROCm/MIOpen/tree/GTests_Refactoring_Integration_Branch)` ?

Yes it is. We decided to use separate branch for our GTests refactoring efforts until we collect all necessary changes. When we collect them we will merge this branch to the develop

**Comment by Vsevolod1983** [comment]
I converted this PR into draft because this PR locks up on CI. I expect this is because of time out of any kind happens. env variables are removed in this PR and it probably overloads CI.

File: test/gtest/adam.cpp [hunk 1]
```diff
@@ -26,23 +26,10 @@
 #include <miopen/env.hpp>
```
Comment (averinevg) [inline test/gtest/adam.cpp]:
```suggestion
```

File: test/gtest/adam.cpp [hunk 2]
```diff
@@ -26,23 +26,10 @@
...
 #include "adam.hpp"
 
-MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
-MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
-
 namespace env = miopen::env;
```
Comment (averinevg) [inline test/gtest/adam.cpp]:
```suggestion
```

File: test/gtest/addlayernorm.cpp [hunk 3]
```diff
@@ -27,21 +27,8 @@
 #include "addlayernorm.hpp"
 #include <miopen/env.hpp>
```
Comment (averinevg) [inline test/gtest/addlayernorm.cpp]:
```suggestion
```

File: test/gtest/cat.cpp [hunk 4]
```diff
@@ -26,23 +26,10 @@
 #include <miopen/env.hpp>
```
Comment (averinevg) [inline test/gtest/cat.cpp]:
```suggestion
```

File: test/gtest/cat.cpp [hunk 5]
```diff
@@ -26,23 +26,10 @@
...
 #include "cat.hpp"
 
-MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
-MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
-
 namespace env = miopen::env;
```
Comment (averinevg) [inline test/gtest/cat.cpp]:
```suggestion
```

File: test/gtest/conv_3d.cpp [hunk 6]
```diff
@@ -30,9 +30,6 @@
 #include <miopen/env.hpp>
```
Comment (averinevg) [inline test/gtest/conv_3d.cpp]:
```suggestion
```

File: test/gtest/conv_extra.cpp [hunk 7]
```diff
@@ -46,38 +44,17 @@ void GetArgs(const std::string& param, std::vector<std::string>& tokens)
...
 }
 
-std::vector<std::string> GetTestCases(void)
+std::vector<std::string> GetTestCases(const std::string& float_arg)
 {
-    const std::string cmd               = "test_conv2d";
-    const std::string cmd_v             = cmd + " --verbose";
+    const std::string cmd               = "test_conv2d ";
+    const std::string cmd_v             = cmd + "--verbose";
     const std::string default_float_arg = " --float";
```
Comment (averinevg) [inline test/gtest/conv_extra.cpp:51]:
```suggestion
```

**Review by averinevg — CHANGES_REQUESTED** [review]

File: test/gtest/conv_for_implicit_gemm.cpp [hunk 8]
```diff
@@ -34,31 +34,19 @@
 
 using TestCase = std::tuple<std::vector<std::string>, std::string>;
 
-MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
 MIOPEN_DECLARE_ENV_VAR_BOOL(IMPLICITGEMM_TESTING_ENV)
-MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
 
 namespace env = miopen::env;
```
Comment (averinevg) [inline test/gtest/conv_for_implicit_gemm.cpp:39]:
```suggestion
```

File: test/gtest/conv_group.cpp [hunk 9]
```diff
@@ -30,9 +30,6 @@
 #include <miopen/env.hpp>
```
Comment (averinevg) [inline test/gtest/conv_group.cpp]:
```suggestion
```

File: test/gtest/conv_extra.cpp (hunk 7 above)
Comment (Vsevolod1983) [reply test/gtest/conv_extra.cpp:51]: it is still used

File: test/gtest/conv_for_implicit_gemm.cpp (hunk 8 above)
Comment (Vsevolod1983) [reply test/gtest/conv_for_implicit_gemm.cpp:39]: it is still used

**Review by averinevg — APPROVED** [review]

**Comment by junliume** [comment]
> > Is the target branch `[GTests_Refactoring_Integration_Branch](https://github.com/ROCm/MIOpen/tree/GTests_Refactoring_Integration_Branch)` ?
> 
> Yes it is. We decided to use separate branch for our GTests refactoring efforts until we collect all necessary changes. When we collect them we will merge this branch to the develop

Maybe update the target branch with merging the latest develop branch will help with the CI. Thanks!

**Comment by Vsevolod1983** [comment]
> > > Is the target branch `[GTests_Refactoring_Integration_Branch](https://github.com/ROCm/MIOpen/tree/GTests_Refactoring_Integration_Branch)` ?
> > 
> > 
> > Yes it is. We decided to use separate branch for our GTests refactoring efforts until we collect all necessary changes. When we collect them we will merge this branch to the develop
> 
> Maybe update the target branch with merging the latest develop branch will help with the CI. Thanks!

yeah, I update it periodically.

Btw, we would like to merge PRs into this temporary branch by ourselves. No objections from you ?

**Review by junliume — APPROVED** [review]

**Pull request merged**